### PR TITLE
[docker] add more metadata labels

### PR DESCRIPTION
**Pull request by v1v**
Similar to https://github.com/jenkinsci/docker/pull/965 to standardise the metadata labels

## Questions

- I don't know the push process for the linux docker images. If it's not within this repo then the BUILD_DATE args should be verified

### CODE REVIEW

File: 11/buster/Dockerfile [hunk 1]
```diff
@@ -30,7 +30,21 @@ ARG gid=1000
...
+# metadata labels
+LABEL \
+    org.label-schema.build-date=$BUILD_DATE \
+    org.label-schema.schema-version="1.0" \
+    org.label-schema.vendor="Jenkins project" \
+    org.label-schema.name="Official Jenkins Docker image" \
+    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (slave.jar)" \
+    org.label-schema.version="${VERSION}" \
+    org.label-schema.url="https://jenkins.io/" \
+    org.label-schema.vcs-url="https://github.com/jenkinsci/jenkins" \
```
Comment (oleg-nenashev) [inline 11/buster/Dockerfile:46]: Wrong URL

File: 11/alpine/Dockerfile [hunk 2]
```diff
@@ -30,7 +30,21 @@ ARG gid=1000
...
+# metadata labels
+LABEL \
+    org.label-schema.build-date=$BUILD_DATE \
+    org.label-schema.schema-version="1.0" \
+    org.label-schema.vendor="Jenkins project" \
+    org.label-schema.name="Official Jenkins Docker image" \
+    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (slave.jar)" \
+    org.label-schema.version="${VERSION}" \
+    org.label-schema.url="https://jenkins.io/" \
+    org.label-schema.vcs-url="https://github.com/jenkinsci/jenkins" \
```
Comment (slide) [inline 11/alpine/Dockerfile:46]:
```suggestion
    org.label-schema.vcs-url="https://github.com/jenkinsci/docker-agent" \
```

File: 11/buster/Dockerfile (hunk 1 above)
Comment (slide) [inline 11/buster/Dockerfile:46]:
```suggestion
    org.label-schema.vcs-url="https://github.com/jenkinsci/docker-agent" \
```

File: 11/buster/Dockerfile [hunk 3]
```diff
@@ -30,7 +30,21 @@ ARG gid=1000
...
+# Build-time metadata as defined at http://label-schema.org
+ARG BUILD_DATE
+
+# metadata labels
+LABEL \
+    org.label-schema.build-date=$BUILD_DATE \
+    org.label-schema.schema-version="1.0" \
+    org.label-schema.vendor="Jenkins project" \
+    org.label-schema.name="Official Jenkins Docker image" \
+    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (slave.jar)" \
```
Comment (slide) [inline 11/buster/Dockerfile:43]:
```suggestion
    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (agent.jar)" \
```

File: 11/stretch/Dockerfile [hunk 4]
```diff
@@ -30,7 +30,21 @@ ARG gid=1000
...
+# Build-time metadata as defined at http://label-schema.org
+ARG BUILD_DATE
+
+# metadata labels
+LABEL \
+    org.label-schema.build-date=$BUILD_DATE \
+    org.label-schema.schema-version="1.0" \
+    org.label-schema.vendor="Jenkins project" \
+    org.label-schema.name="Official Jenkins Docker image" \
+    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (slave.jar)" \
```
Comment (slide) [inline 11/stretch/Dockerfile:43]:
```suggestion
    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (agent.jar)" \
```

File: 11/stretch/Dockerfile [hunk 5]
```diff
@@ -30,7 +30,21 @@ ARG gid=1000
...
+    org.label-schema.version="${VERSION}" \
+    org.label-schema.url="https://jenkins.io/" \
+    org.label-schema.vcs-url="https://github.com/jenkinsci/jenkins" \
```
Comment (slide) [inline 11/stretch/Dockerfile:46]:
```suggestion
    org.label-schema.vcs-url="https://github.com/jenkinsci/docker-agent" \
```

File: 11/windows/nanoserver-1809/Dockerfile [hunk 6]
```diff
@@ -45,7 +45,21 @@ RUN [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tl
...
+# Build-time metadata as defined at http://label-schema.org
+ARG BUILD_DATE
+
+# metadata labels
+LABEL `
+    org.label-schema.build-date=$BUILD_DATE `
+    org.label-schema.schema-version="1.0" `
+    org.label-schema.vendor="Jenkins project" `
+    org.label-schema.name="Official Jenkins Docker image" `
+    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (slave.jar)" `
```
Comment (slide) [inline 11/windows/nanoserver-1809/Dockerfile:58]:
```suggestion
    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (agent.jar)" `

```

File: 11/windows/nanoserver-1809/Dockerfile [hunk 7]
```diff
@@ -45,7 +45,21 @@ RUN [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tl
...
+    org.label-schema.version="${VERSION}" `
+    org.label-schema.url="https://jenkins.io/" `
+    org.label-schema.vcs-url="https://github.com/jenkinsci/jenkins" `
```
Comment (slide) [inline 11/windows/nanoserver-1809/Dockerfile:61]:
```suggestion
    org.label-schema.vcs-url="https://github.com/jenkinsci/docker-agent" `

```

File: 11/windows/windowsservercore-1809/Dockerfile [hunk 8]
```diff
@@ -27,7 +27,21 @@ FROM adoptopenjdk:11-jdk-hotspot-windowsservercore-1809
...
+# Build-time metadata as defined at http://label-schema.org
+ARG BUILD_DATE
+
+# metadata labels
+LABEL `
+    org.label-schema.build-date=$BUILD_DATE `
+    org.label-schema.schema-version="1.0" `
+    org.label-schema.vendor="Jenkins project" `
+    org.label-schema.name="Official Jenkins Docker image" `
+    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (slave.jar)" `
```
Comment (slide) [inline 11/windows/windowsservercore-1809/Dockerfile:40]:
```suggestion
    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (agent.jar)" `

```

File: 11/windows/windowsservercore-1809/Dockerfile [hunk 9]
```diff
@@ -27,7 +27,21 @@ FROM adoptopenjdk:11-jdk-hotspot-windowsservercore-1809
...
+    org.label-schema.version="${VERSION}" `
+    org.label-schema.url="https://jenkins.io/" `
+    org.label-schema.vcs-url="https://github.com/jenkinsci/jenkins" `
```
Comment (slide) [inline 11/windows/windowsservercore-1809/Dockerfile:43]:
```suggestion
    org.label-schema.vcs-url="https://github.com/jenkinsci/docker-agent" `

```

File: 8/alpine/Dockerfile [hunk 10]
```diff
@@ -30,7 +30,21 @@ ARG gid=1000
...
+# Build-time metadata as defined at http://label-schema.org
+ARG BUILD_DATE
+
+# metadata labels
+LABEL \
+    org.label-schema.build-date=$BUILD_DATE \
+    org.label-schema.schema-version="1.0" \
+    org.label-schema.vendor="Jenkins project" \
+    org.label-schema.name="Official Jenkins Docker image" \
+    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (slave.jar)" \
```
Comment (slide) [inline 8/alpine/Dockerfile:43]:
```suggestion
    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (agent.jar)" \
```

File: 8/alpine/Dockerfile [hunk 11]
```diff
@@ -30,7 +30,21 @@ ARG gid=1000
...
+    org.label-schema.version="${VERSION}" \
+    org.label-schema.url="https://jenkins.io/" \
+    org.label-schema.vcs-url="https://github.com/jenkinsci/jenkins" \
```
Comment (slide) [inline 8/alpine/Dockerfile:46]:
```suggestion
    org.label-schema.vcs-url="https://github.com/jenkinsci/docker-agent" \
```

File: 8/buster/Dockerfile [hunk 12]
```diff
@@ -30,7 +30,21 @@ ARG gid=1000
...
+# Build-time metadata as defined at http://label-schema.org
+ARG BUILD_DATE
+
+# metadata labels
+LABEL \
+    org.label-schema.build-date=$BUILD_DATE \
+    org.label-schema.schema-version="1.0" \
+    org.label-schema.vendor="Jenkins project" \
+    org.label-schema.name="Official Jenkins Docker image" \
+    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (slave.jar)" \
```
Comment (slide) [inline 8/buster/Dockerfile:43]:
```suggestion
    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (agent.jar)" \
```

File: 8/buster/Dockerfile [hunk 13]
```diff
@@ -30,7 +30,21 @@ ARG gid=1000
...
+    org.label-schema.version="${VERSION}" \
+    org.label-schema.url="https://jenkins.io/" \
+    org.label-schema.vcs-url="https://github.com/jenkinsci/jenkins" \
```
Comment (slide) [inline 8/buster/Dockerfile:46]:
```suggestion
    org.label-schema.vcs-url="https://github.com/jenkinsci/docker-agent" \
```

File: 8/stretch/Dockerfile [hunk 14]
```diff
@@ -30,7 +30,21 @@ ARG gid=1000
...
+# Build-time metadata as defined at http://label-schema.org
+ARG BUILD_DATE
+
+# metadata labels
+LABEL \
+    org.label-schema.build-date=$BUILD_DATE \
+    org.label-schema.schema-version="1.0" \
+    org.label-schema.vendor="Jenkins project" \
+    org.label-schema.name="Official Jenkins Docker image" \
+    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (slave.jar)" \
```
Comment (slide) [inline 8/stretch/Dockerfile:43]:
```suggestion
    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (agent.jar)" \
```

File: 8/stretch/Dockerfile [hunk 15]
```diff
@@ -30,7 +30,21 @@ ARG gid=1000
...
+    org.label-schema.version="${VERSION}" \
+    org.label-schema.url="https://jenkins.io/" \
+    org.label-schema.vcs-url="https://github.com/jenkinsci/jenkins" \
```
Comment (slide) [inline 8/stretch/Dockerfile:46]:
```suggestion
    org.label-schema.vcs-url="https://github.com/jenkinsci/docker-agent" \
```

File: 8/windows/nanoserver-1809/Dockerfile [hunk 16]
```diff
@@ -45,7 +45,21 @@ RUN [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tl
...
+# Build-time metadata as defined at http://label-schema.org
+ARG BUILD_DATE
+
+# metadata labels
+LABEL `
+    org.label-schema.build-date=$BUILD_DATE `
+    org.label-schema.schema-version="1.0" `
+    org.label-schema.vendor="Jenkins project" `
+    org.label-schema.name="Official Jenkins Docker image" `
+    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (slave.jar)" `
```
Comment (slide) [inline 8/windows/nanoserver-1809/Dockerfile:58]:
```suggestion
    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (agent.jar)" `

```

File: 8/windows/nanoserver-1809/Dockerfile [hunk 17]
```diff
@@ -45,7 +45,21 @@ RUN [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tl
...
+    org.label-schema.version="${VERSION}" `
+    org.label-schema.url="https://jenkins.io/" `
+    org.label-schema.vcs-url="https://github.com/jenkinsci/jenkins" `
```
Comment (slide) [inline 8/windows/nanoserver-1809/Dockerfile:61]:
```suggestion
    org.label-schema.vcs-url="https://github.com/jenkinsci/docker-agent" `

```

File: 8/windows/windowsservercore-1809/Dockerfile [hunk 18]
```diff
@@ -27,7 +27,21 @@ FROM adoptopenjdk:8-jdk-hotspot-windowsservercore-1809
...
+# Build-time metadata as defined at http://label-schema.org
+ARG BUILD_DATE
+
+# metadata labels
+LABEL `
+    org.label-schema.build-date=$BUILD_DATE `
+    org.label-schema.schema-version="1.0" `
+    org.label-schema.vendor="Jenkins project" `
+    org.label-schema.name="Official Jenkins Docker image" `
+    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (slave.jar)" `
```
Comment (slide) [inline 8/windows/windowsservercore-1809/Dockerfile:40]:
```suggestion
    org.label-schema.description="This is a base image, which provides the Jenkins agent executable (agent.jar)" `
```

File: 8/windows/windowsservercore-1809/Dockerfile [hunk 19]
```diff
@@ -27,7 +27,21 @@ FROM adoptopenjdk:8-jdk-hotspot-windowsservercore-1809
...
+    org.label-schema.version="${VERSION}" `
+    org.label-schema.url="https://jenkins.io/" `
+    org.label-schema.vcs-url="https://github.com/jenkinsci/jenkins" `
```
Comment (slide) [inline 8/windows/windowsservercore-1809/Dockerfile:43]:
```suggestion
    org.label-schema.vcs-url="https://github.com/jenkinsci/docker-agent" `
```

**Comment by slide** [comment]
@v1v The Linux image build is done on Dockerhub, which means a BUILD_DATE arg may not be doable for the official images.

**Comment by v1v** [comment]
> @v1v The Linux image build is done on Dockerhub, which means a BUILD_DATE arg may not be doable for the official images.

I see, I like the idea of having the env variable to be injected if possible or even being created on the fly within the docker image, or maybe something else, @slide, what approach do you think it might be a good one for this particular case?

**Comment by slide** [comment]
If we want full control over the builds, thus allowing us to inject variables and so forth (which would be nice in many circumstances), we would need to update the Jenkinsfile to do the build and push on our CI infra instead of dockerhub and disable the dockerhub builds. @oleg-nenashev  and @MarkEWaite would be good to have weigh in on this as well.

**Comment by v1v** [comment]
I just removed all the references to the `BUILD_DATE` https://github.com/jenkinsci/docker-agent/pull/141/commits/aacf86355529b5793283e76d263f806409dc94c8

**Comment by timja** [comment]
@v1v would you be able to resolve the conflicts please then we can ship this?

**Comment by timja** [comment]
Looks like opencontainers is the replacement: https://github.com/opencontainers/image-spec/blob/master/annotations.md

**Comment by timja** [comment]
Replaced by https://github.com/jenkinsci/docker-agent/pull/200

**Pull request closed without merging**